### DISCUSSION TITLE
Azure Pipelines Scaler

### DISCUSSION THREAD

**Discussion by cirux91** [kedacore/keda#2674] [Q&A / Need Help]
I Observed that Keda Azure Pipeline Scaler doesn't work when an agent pool contains multiple agents. 

In my case I've an agent pool with 4 self-hosted agents deployed on AKS. I'd like to have the the possiblity to scale each agent when a new pipeline request it.

Thanks in advance.

**Comment by tomkerkhove** [comment]
Hm, I'm not sure I get the scenario - Are those 4 agents part of the same pool? If so, why would you want to scale them all for every pipeline? Is it because the pipeline has multiple jobs? If so, then these agents are port of different pools, no?

**Reply by cirux91** [reply]
Hi @tomkerkhove thank u for your answer. Let me explain better: as you can see into screenshot I've 4 agents in the same pool and they  don't scale all for every pipeline. So my question... in this case 1 pool and 4 agents the keda plugin still working? Into doc it seems that keda works only with 1 agent for pool. 

![image](https://user-images.githubusercontent.com/4045663/155949907-fbf06028-3497-415a-a222-28f4922e84b5.png)

**Reply by tomkerkhove** [reply]
@JorTurFer is more of an expert on this - We should have support for it though, if possible.

**Reply by JorTurFer** [reply]
Hey!
I guess that in your scenario, you have those agents with their own capabilities, right? I base of the required capabilities for each job, one agent or another takes the job.
Unluckily KEDA doesn't support it right now, but luckily there is [an open issue](https://github.com/kedacore/keda/issues/2328) for it and there is activity there.
Meanwhile, there are some workaround possible, the easiest could be scale all the agents at once (using one SO for each) or spread into different pools.
